backend/src/user-queries/user-queries.controller.ts

```diff
@@ -33,6 +33,14 @@ export class UserQueriesController {
     return this.userQueriesService.findAllForDatabase(databaseId);
   }
 
+  @Post("/parse/:databaseId")
+  parsePipeline(
+    @Param("databaseId") databaseId: string,
+    @Body() pipeline: Pipeline,
```

Comment:
We should decide one one way to pass around user-queries, currently when we "run" we only pass the ID, assuming its been saved beforehand. If we matched that here, we'd pass the user-query id instead of the pipeline itself.

Lets decide whether we want to standardize

Reply:
We hit this endpoint before the pipeline is saved to the user query. This endpoint just parses the pipeline into sql, that sql is then used to determine if the pipeline differs from the sql in UI.